geode-core/src/main/java/org/apache/geode/internal/cache/backup/BackupManager.java

```diff
@@ -106,65 +109,86 @@ public void validateRequestingAdmin() {
       if (abort) {
         return new HashSet<>();
       }
-      HashSet<PersistentID> persistentIds = new HashSet<>();
+      tempDirectory = Files.createTempDirectory("backup_" + System.currentTimeMillis());
```

Comment:
My understanding is that each member backing up its disk stores now copies all the file to this "tempDirectory". This seems like a big change that could cause customers to need much more disk space during a backup. Even if we use hard links to move the files to this directory, hard links are not always available and it will then need to copy the file twice (once to the temp dir and once to the destination).

Why is a tempDirectory needed? Can we do a backup without it?

Reply:
The temporary directory is required because we now do not backup the Oplog files to their final destination while holding locks that prevent Oplog deletion. It also results in the user, config files, and deployed jars being saved in the backup in the state they are in closest to the backup, instead of eventually being copied (for example, in a future backup to cloud storage, where there would be a delay between the backup being "done" and transported to the storage provider.

This method does not inherently use more space during the backup than we did before (recently) adding the hard-link support, as we copied the oplog files to where the backup was conducted. Now, if their backup location is remote from their local filesystem and the temporary location is on a different filesystem than the disk stores (preventing hard-links), it will use more space on the local file system for the duration of the backup.

Reply:
This temporary directory change seems like a big change to how backup is done. If we could make sure that if the oplogs are on multiple file systems that we would copy them to a temp directory on the same file system then I'd accept this change.

Reply:
One option is to make a directory temporarily within the directory for each diskstore to hold the files for that diskstore during the backup. That should remove the likelihood that oplogs would need to be moved between filesystems.